credentialdigger/client.py

```diff
@@ -1006,3 +1010,25 @@ def _get_scan_rules(self, category=None, exclude=None):
             raise ValueError('No rules found')
 
         return rules
+
+    def update_similar_snippets(self,
+                                target_snippet,
+                                state,
+                                repo_url,
+                                file_name=None,
```

Comment:
why do you need the filename? As far as I understood, given a `target_snippet` you need to get all the "new" discoveries for a certain repo, and then compute each similarity and eventually flag other FPs. So what's the need of having the file name in this process?

Reply:
The purpose of having a filename here is to allow for exclusive updating of similar snippets found in the same file as the target snippet, because there could be a situation where a file contains similar snippets as another file, but one with real passwords and the other with dummy ones. This parameter is provided 'just in case' someone needs this feature, but is not meant to be used as a general rule. I can remove it if you think it should not be there.

Reply:
Let's keep it with this PR then, but before merging into the main branch I would like to test it. Moreover, I'm not fully convinced this parameter is really needed: if the same snippet appears in 2 different files, then either they are treated at the same way (i.e., both TP or both FP), or it is the path model that should flag one of the two as a FP.

Reply:
Yes in theory, but then why need a snippet model if the path model is always right? Some snippets with fake credentials can appear in paths that aren't dedicated to testing and/or detected as dummy by the path model, since we're applying a model to distinguish real from fake passwords after the path model. The snippet model may fail to recognize a fake password as being fake, and that's the whole point of having a similarity model: to provide a simple, 'user-powered' way of compensating for failings/shortcomings of one of the two models. This may become superfluous when the snippet model reaches a very high accuracy, but seems quite relevant as of now. Adding the filename parameter offers a more granular control over the update, which can't be a bad thing. In the UI, it could just take the form of a button to tick 'restrict updates to current file'. I noticed, there are often similar snippets in the same file, so it would still be efficient using it that way.

Reply:
(There clearly is a (very) small danger bound to the similarity feature, and steps to mitigate this danger should be taken, I think. Otherwise, we'll need to clearly inform the user on the potential risks of overly using this functionality on snippets that do contain a password).